lib/puppet/provider/foreman_hostgroup/rest_v3.rb

```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
+
+Puppet::Type.type(:foreman_hostgroup).provide(
+  :rest_v3,
+  parent: Puppet::Type.type(:foreman_resource).provider(:rest_v3)
+) do
+  confine feature: %i[json oauth]
+
+  def initialize(value = {})
+    super
+    @property_flush = {}
+  end
+
+  def exists?
+    !id.nil?
+  end
+
+  def create
+    Puppet.debug("Creating Foreman Hostgroup #{resource[:name]} with parent #{resource[:parent_hostgroup]}")
+
+    if resource[:parent_hostgroup] && parent_hostgroup_id.nil?
+      raise Puppet::Error,
+            "Parent hostgroup #{resource[:parent_hostgroup]} for #{resource[:name]} not found"
+    end
+
+    organization_ids = (resource[:organizations]&.map { |org| organization_id(org) })
+    location_ids = (resource[:locations]&.map { |loc| location_id(loc) })
```

Comment:
Technically the braces aren't needed
```suggestion
    organization_ids = resource[:organizations]&.map { |org| organization_id(org) }
    location_ids = resource[:locations]&.map { |loc| location_id(loc) }
```

Reply:
Brackets removed